.github/workflows/docker-hub.yml

```diff
@@ -0,0 +1,44 @@
+name: Docker Hub Workflow
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - 'main'
+    tags:
+      - 'v*'
```

Comment:
Is is still necessary? 
Is it for staging, right? We want all the branches merged to main to be pushed to dockerhub, isn't it ?

Reply:
I might be wrong, but this job will be used to push Docker Images, in any situation. It should run when:
- pushes to the 'main' branch
- pushes to tags starting with 'v'
- pull requests targeting the 'main' branch (to run Docker build without push).